### PR TITLE
Bump the supported solidity version range

### PR DESCRIPTION
**Pull request by fvictorio**

### CODE REVIEW

File: packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts [hunk 1]
```diff
@@ -814,6 +823,18 @@ describe("Stack traces", function () {
...
 });
 
+describe("Solidity support", function () {
+  it("check that the latest tested version is within the supported version range", async function () {
+    const latestSupportedVersion = solidityCompilers
+      .map((sc) => sc.solidityVersion)
+      .sort(semver.compare)[solidityCompilers.length - 1];
+
+    assert.isTrue(
+      semver.satisfies(latestSupportedVersion, SUPPORTED_SOLIDITY_VERSION_RANGE)
```
Comment (fvictorio) [inline packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:833]: This doesn't check that the range is tight. For example, if we increment it to `<=0.8.5` now, this test will continue to pass. AFAIK there's no way to do this with the `semver` module, but something we can do is to add some tests that bump the patch, minor and major version of `lastSupportedVersion` and assert that they are **not** within the range.

Comment (alcuadrado) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:833]: We can add it by bumping a minor and the next major's `x.0.0`. That should cover most cases.

Comment (fvictorio) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:833]: Why wouldn't we do it for the minor too? It's way more likely than a next major (I don't know when there will be a solidity 1.0.0) and it's just an extra line. Am I missing something here?

Comment (alcuadrado) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:833]: Sorry, I meant patch. Also, i always forget that solc versions are always `0.x.y`.

We can do this. If  `x.y.z` is the latest one, we can test with `x+1.0.0`, `x.y+1.0`, and `x.y.z+1`

Comment (fvictorio) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:833]: That's my original idea, I must explained it really badly :sweat_smile: I'll do that then!

Comment (fvictorio) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts]: 44c1baee50e5d5a0251ffdef75c930601379c9d2

File: packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts [hunk 2]
```diff
@@ -814,6 +823,39 @@ describe("Stack traces", function () {
...
+
+    assert.isTrue(
+      semver.satisfies(
+        latestSupportedVersion,
+        SUPPORTED_SOLIDITY_VERSION_RANGE
+      ),
+      `Expected ${latestSupportedVersion} to be within the ${SUPPORTED_SOLIDITY_VERSION_RANGE} range`
+    );
+
+    const nextPatchVersion = semver.inc(latestSupportedVersion, "patch")!;
```
Comment (alcuadrado) [inline packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts:840]: Oh, this was easier than expected! Nice!

**Review by alcuadrado — APPROVED** [review]

**Pull request merged**